src/transformers/image_transforms.py

```diff
@@ -278,6 +289,115 @@ def get_resize_output_image_size(
     return (new_long, new_short) if width <= height else (new_short, new_long)
 
 
+def colorize_depth(
+    value: Union[np.ndarray, "torch.Tensor"],
+    vmin: Optional[Union[float, None]] = None,
+    vmax: Optional[Union[float, None]] = None,
+    vmin_perc: Optional[float] = 1.0,
+    vmax_perc: Optional[float] = 99.0,
+    cmap: Optional[str] = "gray_r",
+    invalid_val: Optional[int] = -99,
+    invalid_mask: Optional[Union[np.ndarray, None]] = None,
+    background_color: Optional[Tuple[int]] = (128, 128, 128, 255),
+    gamma_corrected: Optional[bool] = False,
+    normalize: Optional[bool] = False,
+    value_transform: Optional[Union[Callable, None]] = None,
+    return_numpy: bool = False,
+) -> Union[np.ndarray, "PIL.Image.Image"]:
+    """Converts a depth map to a color image.
+
+    Args:
+        value (`torch.Tensor` or `numpy.ndarray`):
+            Input depth map. Shape: (H, W) or (1, H, W) or (1, 1, H, W). All singular dimensions are squeezed
+        vmin (`float`, *optional*):
+            vmin-valued entries are mapped to start color of cmap. If None, value.min() is used.
+        vmax (`float`, *optional*):
+            vmax-valued entries are mapped to end color of cmap. If None, value.max() is used.
+        vmin_perc (`float`, *optional*, defaults to `1.0`):
+            use the `vmin_perc`-th percentile as `vmin` (outlier rejection).
+        vmax_perc (`float`, *optional*, defaults to `99.0`):
+            use the `vmax_perc`-th percentile as `vmax` (outlier rejection).
+        normalize (`bool`, *optional*, defaults to `False`):
+            Apply normalization between [0,1] for the colored image values.
+        cmap (`str`, *optional*, defaults to `gray_r`):
+            matplotlib colormap to use (requires matplotlib).
+        invalid_val (`int`, *optional*, defaults to `-99`):
+            Specifies value of invalid pixels that should be colored as 'background_color'.
+        invalid_mask (`numpy.ndarray`, *optional*):
+            Boolean mask for invalid regions.
+        background_color (`Tuple[int]`, *optional*, defaults to `(128, 128, 128, 255)`):
+            4-tuple RGB color to give to invalid pixels.
+        gamma_corrected (`bool`, *optional*, defaults to `False`):
+            Apply gamma correction to colored image.
+        value_transform (Callable, *optional*):
+            Apply transform function to valid pixels before coloring.
+        return_numpy (`bool`, *optional*, defaults to `False`):
+            Whether or not to return the resized image as a numpy array. If False a `PIL.Image.Image` object is
+            returned.
```

Comment:
Considering that the inputs to this function will not be used again in a torch-specific way since they are only processed here to be visualized, doesn't it make sense to always output numpy arrays and, if available, PIL images?

Reply:
No, the purpose of post-processing is to perform model-specific transformations which standardize the outputs for models for specific tasks e.g. making sure bounding boxes are in the same format, it's not to provide utility functionality for things like visualization.

Reply:
In that case, does this mean it should not even return the `depth` key on the dictionary, i.e. just remove any padding and resize etc?

Reply:
okay, in that case the `colorize_depth` function will not be used at all. So if we don't need it for something else, I will delete it

Reply:
sounds good - let's kill our darlings :)